testing/fil_token_integration/actors/basic_token_actor/src/lib.rs

```diff
@@ -112,12 +112,36 @@ impl FrcXXXToken<RuntimeError> for BasicToken<'_> {
 
 impl BasicToken<'_> {
     fn mint(&mut self, params: MintParams) -> Result<MintReturn, RuntimeError> {
-        self.util.mint(
+        // mint shouldn't save state at all
+        let old_cid = sdk::sself::root().unwrap();
+        let receiver_params = self.util.mint(
             &caller_address(),
             &params.initial_owner,
             &params.amount,
             &Default::default(),
         )?;
+        
+        let cid = self.util.flush()?;
+        sdk::sself::set_root(&cid).unwrap();
+
+        /*
+        let param_bytes = fvm_ipld_encoding::to_vec(&receiver_params);
+        let message = format!("receiver hook params: {:?}", param_bytes);
+        sdk::vm::abort(fvm_shared::error::ExitCode::USR_ILLEGAL_ARGUMENT.value(),
+                        Some(message.as_str()));
+        */
+
+        if let Err(e) = self.util.call_receiver_hook(&params.initial_owner, receiver_params) {
+            // revert to previous on-chain state
+            sdk::sself::set_root(&old_cid).unwrap();
```

Comment:
We shouldn't need to do this so long as invoke will abort if this function returns an error. I think that's a reasonable API – if any method returns an error, invoke will abort

Reply:
Makes sense, I was thinking about restoring things in order to take other actions (like calling another actor) upon failure. But the more I think about that kind of use, the more it feels like it belongs in a separate actor. Easier at that level to call a method like mint or transfer on the token actor and then take further action based on the result. Also makes this code a lot simpler because it can just return a `Result` and we'll ultimately handle that in `invoke`.